packages/schema/src/model.ts

```diff
@@ -16,25 +16,69 @@ import {
   IJupyterCadModel,
   ISelection,
   IUserData,
-  Pointer
+  Pointer,
+  IJCadSettings
 } from './interfaces';
 import jcadSchema from './schema/jcad.json';
 import { Contents } from '@jupyterlab/services';
+import { ISettingRegistry } from '@jupyterlab/settingregistry';
+
+const SETTINGS_ID = '@jupytercad/jupytercad-core:jupytercad-settings';
 
 export class JupyterCadModel implements IJupyterCadModel {
   constructor(options: JupyterCadModel.IOptions) {
-    const { annotationModel, sharedModel } = options;
+    const { annotationModel, sharedModel, settingRegistry } = options;
     if (sharedModel) {
       this._sharedModel = sharedModel;
     } else {
       this._sharedModel = this.createSharedModel();
     }
     this._connectSignal();
     this.annotationModel = annotationModel;
+    this.settingRegistry = settingRegistry;
     this._copiedObject = null;
     this._pathChanged = new Signal<JupyterCadModel, string>(this);
   }
 
+  /**
+   * Initialize custom settings for JupyterLab.
+   */
+  readonly settingsChanged = new Signal<this, void>(this);
```

Comment:
you should expose `ISignal` property only, the real signal property should be private

Reply:
Done in [a7bead7](https://github.com/jupytercad/JupyterCAD/pull/725/commits/a7bead78d3d3202931b0550f56e7210becc57859)